### PR TITLE
Api

### PR DESCRIPTION
**Pull request by j-waters**
Depends on #106 

Currently just have an endpoint for everything we log from #106.

Features that I should probably add:

 <s>Pagination</s> - I don't think it's really necessary at the moment
- [x] Filtering
- [x] Live updates (Server Sent Events? Websockets?)

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
+
+python crypto_trading.py &
+gunicorn api_server:app -w 1 --threads 1 -b 0.0.0.0:8000
```
Comment (ryan-shaw) [inline entrypoint.sh]: Might be better to add something like supervisorctrl/s6-overlay to deal with this, if crypto_trading.py crashes the container will continue as the gunicorn process is in the foreground. 

Better yet- run one service per container and add a second container to run the gunicorn

Comment (j-waters) [reply entrypoint.sh]: Completely forgot we were using docker compose good shout I've added a separate container

**Review by AnasAlmomany — APPROVED** [review]
Great work man, Clean code is always appreciated

File: docker-compose.yml [hunk 2]
```diff
@@ -3,8 +3,20 @@ version: "3"
 services:
   crypto-trading:
     build: .
+    image: binance_trader
     container_name: binance_trader
     volumes:
     - ./user.cfg:/app/user.cfg
```
Comment (aelbozie) [inline docker-compose.yml:8]: I don't think this needs to be in volumes as the whole directory is copied

File: docker-compose.yml [hunk 3]
```diff
@@ -3,8 +3,20 @@ version: "3"
...
-    - ./.current_coin:/app/.current_coin
-    - ./.current_coin_table:/app/.current_coin_table
+    - ./data:/app/data
```
Comment (aelbozie) [inline docker-compose.yml:10]: Exactly this, makes sense to mount a whole directory not individual files

**Review by aelbozie — APPROVED** [review]

**Comment by j-waters** [comment]
I haven't added pagination but I don't think it's necessary for now

**Comment by j-waters** [comment]
Rebased

**Comment by j-waters** [comment]
@edeng23 I'm sure we'll add to the API or change it in the future but would it be okay if it was merged soonish? Don't want to have to keep rebasing it is all!

**Comment by edeng23** [comment]
@j-waters Sure,you're right. I'll start testing the code ASAP.

**Pull request merged**